### PR TITLE
5.x merge 4.x

### PR DESCRIPTION
**Pull request by asmorkalov**
Contrib: https://github.com/opencv/opencv_contrib/pull/3710
Extra: https://github.com/opencv/opencv_extra/pull/1164

#24415 from thewoz:imread
#24724 from tomoaki0705:carotene_warnings
#24877 from zzuliys:feature/mac
#24893 from chacha21:cart_polar_inplace
#25042 from mshabunin:doc-upgrade
#25055 from dmatveev:dm/value_initialized_gmat
#25092 from klatism:libjpeg-upgrade
#25096 from klatism:libtiff-upgrade
#25120 from LaurentBerger:I25103
#25122 from unnonouno:pqueue
#25123 from klatism:zlib-upgrade
#25125 from asmorkalov:as/backport_imgproc_c_removal
#25130 from CSBVision:patch-6
#25141 from Kumataro:fix25140_4.x
#25143 from Aman-Vishwakarma1729:4.x
#25145 from MaximSmolskiy:improve-contours-approximations-in-ChessBoardDetector-generateQuads-2
#25149 from asmorkalov:as/error_cleanup_backport
#25160 from asmorkalov:as/backport_highgui_22754
#25162 from mshabunin:doc-upgrade-fix
#25163 from dkurt:onnx_graph_ref
#25166 from fengyuentau:fix_cann_gemm
#25171 from mshabunin:doc-auto-dot
#25176 from cabelo:4.x
#25177 from MaximSmolskiy:speed-up-adaptive-image-binary-threshold-in-findChessboardCorners
#25180 from mshabunin:doc-fix-javadoc
#25181 from dkurt:release_conv_weights
#25182 from MaximSmolskiy:increase-decomposeProjectionMatrix-precision-for-small-scales
#25183 from MaximSmolskiy:fix-bug-in-maskBoundingRect
#25184 from dkurt:avoid_extra_memset
#25189 from klatism:ios-build-fixes
#25190 from klatism:android-config-flags-enhancement                                                                                                                                                                                                                           
#25199 from olpipi:fix_openvino_2024_0                                                                                                                                                                                                                                         
#25203 from mshabunin:fix-scalable-intrin-test                                                                                                                                                                                                                                 
#25204 from mshabunin:fix-rvv-vne                                                                                                                                                                                                                                              
#25213 from mshabunin:fix-doc-buttons                                                                                                                                                                                                                                          
#25217 from fengyuentau:float16_datatype_renaming
#25229 from giokara-oqton:fix_jpegturbo_win64                                                                                                                                                                                                                                  
#25230 from hanliutong/rvv-conv                                                                                                                                                                                                                                                
#25237 from YusukeKameda:4.x                                                                                                                                                                                                                                                   
#25238 from fengyuentau:optimized_const                                                                                                                                                                                                                                        
#25249 from carabsc:cara_changes                                                                                                                                                                                                                                               
#25255 from Kumataro:refactor_imgcodec_tiff_add_imwrite_tiff                                                                                                                                                                                                                   
#25257 from Kumataro:fix25256                                                                                                                                                                                                                                                  
#25258 from AleksandrPanov:fix_rotated_symmetric_grid                                                                                                                                                                                                                          
#25264 from lpylpy0514:4.x                                                                                                                                                                                                                                                     
#25269 from RoboSchmied:RoboSchmied-AGfixT56                                                                                                                                                                                                                                   
#25271 from fengyuentau:matmul_bias                                                                                                                                                                                                                                            
#25273 from dkurt:tflite_new_layers                                                                                                                                                                                                                                            
#25280 from Kumataro:fix25274                                                                                                                                                                                                                                                  
#25282 from asmorkalov:as/deterministic_undistort_test                                                                                                                                                                                                                         
#25285 from johnteslade:cgroupsv2-support                                                                                                                                                                                                                                      
#25291 from dkurt:einsum_openvino                                                                                                                                                                                                                                              
#25293 from asmorkalov:as/dct_hal_fix                                                                                                                                                                                                                                          
#25294 from asmorkalov:as/InferROI_TestStreamingInfer_supress                                                                                                                                                                                                                  
#25299 from Kumataro:add_fix25274                                                                                                                                                                                                                                              
#25301 from zzuliys:feature/mismatch_hashcode

Previous "Merge 4.x": #25119

```
force_builders=Linux AVX2,Custom
build_image:Docs=docs-js
build_image:Custom=javascript
buildworker:Custom=linux-1,linux-4,linux-f1
```

### CODE REVIEW

**Comment by asmorkalov** [comment]
@mshabunin @opencv-alalek @fengyuentau @dkurt Could you take a look?

File: CMakeLists.txt [hunk 1]
```diff
@@ -332,6 +332,9 @@ OCV_OPTION(WITH_OPENEXR "Include ILM support via OpenEXR" ((WIN32 OR ANDROID OR
 OCV_OPTION(WITH_OPENGL "Include OpenGL support" OFF
   VISIBLE_IF NOT ANDROID AND NOT WINRT
   VERIFY HAVE_OPENGL)
+OCV_OPTION(WITH_OPENVX "Include OpenVX support" OFF
```
Comment (mshabunin) [inline CMakeLists.txt:335]: OpenVX has been removed in 4.x.

File: modules/core/include/opencv2/core/cvdef.h [hunk 2]
```diff
@@ -898,15 +898,15 @@ class float16_t
     #endif
     }
 
-    static float16_t fromBits(ushort b)
```
Comment (mshabunin) [inline modules/core/include/opencv2/core/cvdef.h:901]: These methods has been removed from this class.

Comment (asmorkalov) [reply modules/core/include/opencv2/core/cvdef.h:901]: Removed.

File: CMakeLists.txt (hunk 1 above)
Comment (asmorkalov) [reply CMakeLists.txt]: Removed.

**Review by dkurt — APPROVED** [review]

File: modules/dnn/test/test_layers.cpp [hunk 3]
```diff
@@ -1764,6 +1764,50 @@ INSTANTIATE_TEST_CASE_P(/**/, Layer_Test_ShuffleChannel, Combine(
...
+        MatShape gt_shape;
+        for (int j = 0; j < szData.size(); j++)
+        {
+            if (i == j) continue;
+            gt_shape.push_back(szData[j]);
+        }
+
+        EXPECT_EQ(gt_shape, shape(output));
+
+        Mat a = output.reshape(1, output.total());
```
Comment (fengyuentau) [inline modules/dnn/test/test_layers.cpp:1805]: Unused variable `a`.

File: CMakeLists.txt [hunk 4]
```diff
@@ -570,6 +572,10 @@ if(OPENCV_DISABLE_FILESYSTEM_SUPPORT)
   add_definitions(-DOPENCV_HAVE_FILESYSTEM_SUPPORT=0)
 endif()
 
+# MathJax is used for math rendering by both Doxygen HTML and JavaDoc, so
+# this var have to be defined before "modules" AND "doc" are processed
+set(OPENCV_MATHJAX_RELPATH "https://cdn.jsdelivr.net/npm/mathjax@3.0.1" CACHE STRING "URI to a MathJax installation")
```
[the remark this answers is not in the document]
Comment (asmorkalov) [reply CMakeLists.txt:577]: Dropped redundant OPENCV_MATHJAX_RELPATH setting.

File: modules/3d/include/opencv2/3d.hpp [hunk 5]
```diff
@@ -18,7 +18,7 @@
 /**
   @defgroup _3d 3D vision functionality
 
-Most of the functions in this section use a so-called pinhole camera model. The view of a scene
+The functions in this section use a so-called pinhole camera model. The view of a scene
```
[the remark this answers is not in the document]
Comment (asmorkalov) [reply modules/3d/include/opencv2/3d.hpp]: Reverted changes.

File: modules/dnn/test/test_layers.cpp (hunk 3 above)
Comment (asmorkalov) [reply modules/dnn/test/test_layers.cpp]: Removed.

File: modules/imgproc/test/test_boundingrect.cpp [hunk 6]
```diff
@@ -140,4 +140,21 @@ void CV_BoundingRectTest::run(int)
...
+    {
+        for (int image_height = 1; image_height < 15; image_height++)
+        {
+            cv::Rect rect(0, image_height - 1, 3, 1);
+
+            cv::Mat image(cv::Size(image_width, image_height), CV_8UC1, cv::Scalar(0));
+            image(rect) = 255;
+
+            cv::Rect brect = boundingRect(image);
+            ASSERT_EQ(brect, rect);
```
[the remark this answers is not in the document]
Comment (asmorkalov) [reply modules/imgproc/test/test_boundingrect.cpp]: Reverted.

File: modules/calib/src/calibinit.cpp [hunk 7]
```diff
@@ -582,8 +582,10 @@ bool findChessboardCorners(InputArray image_, Size pattern_size,
         }
```
[the remark this answers is not in the document]
Comment (asmorkalov) [reply modules/calib/src/calibinit.cpp:582]: Removed the line.

**Review by opencv-alalek — APPROVED** [review]

**Pull request merged**